weaviate/collection/grpc.py

```diff
@@ -603,21 +607,46 @@ def near_object_options(
             )
         ]
 
-
-class _GrpcCollection(Generic[Properties], _Grpc[Properties]):
-    def __init__(self, connection: Connection, name: str, tenant: Optional[str]):
-        super().__init__(connection, name, tenant)
-
-    def _result_to_object(self, obj: GrpcResult) -> _Object[Properties]:
+    def _parse_out(self, obj: GrpcResult) -> _RawObject:
+        out: _RawObject = {}
         data = obj.result
         for key in data.keys():
             entry = data[key]
             if isinstance(entry, list):
                 value: List = entry
                 for i, _ in enumerate(value):
                     value[i] = self._result_to_object(value[i])
-        data = cast(Properties, data)
-        return _Object[Properties](data=data, metadata=obj.metadata)
+                out[key] = value
+            else:
+                out[key] = self.__struct_value_to_py_value(entry)
+        return out
+
+    def __struct_value_to_py_value(self, value: _StructValue) -> _PyValue:
+        if isinstance(value, struct_pb2.Struct):
+            return {key: self.__struct_value_to_py_value(value) for key, value in value.items()}
+        elif isinstance(value, struct_pb2.ListValue):
+            return [
+                self.__struct_value_to_py_value(cast(_StructValue, value)) for value in value.values
+            ]
+        elif isinstance(value, str):
+            return value
+        elif isinstance(value, float):
+            return value
+        elif isinstance(value, bool):
+            return value
+        elif value is None:
+            return None
+        else:
+            raise ValueError(f"Unknown type: {type(value)}")
```

Comment:
you could do

```
elif isinstance(value, str) or isinstance(value, float) or isinstance(value, bool):
   return value
else:
    assert value is None
    return None
```

Reply:
Does this help with codecov and/or typing?

Reply:
the first change (elif isinstance(value, str) or isinstance(value, float) or isinstance(value, bool)) just saves some lines and groups things together.

Second helps with coverage as we will never go into the else part otherwise. But I don't feel strongly about it